pkg/ipam/ipam.go

```diff
@@ -49,52 +47,15 @@ const (
 
 // NewIPAM returns a new IP address manager
 func NewIPAM(nodeAddressing datapath.NodeAddressing) *IPAM {
-	ipamSubnets := net.IPNet{
-		IP:   nodeAddressing.IPv6().Router(),
-		Mask: defaults.StateIPv6Mask,
-	}
-
 	ipam := &IPAM{
 		nodeAddressing: nodeAddressing,
-		IPAMConfig: allocator.IPAMConfig{
-			Name: "cilium-local-IPAM",
-			Range: &allocator.Range{
-				Subnet:  cniTypes.IPNet(ipamSubnets),
-				Gateway: nodeAddressing.IPv6().Router(),
-			},
-			Routes: []*cniTypes.Route{
-				// IPv6
-				{
-					Dst: net.IPNet{
-						IP:   nodeAddressing.IPv6().Router(),
-						Mask: net.CIDRMask(128, 128),
-					},
-				},
-				{
-					Dst: defaults.IPv6DefaultRoute,
-					GW:  nodeAddressing.IPv6().Router(),
-				},
-			},
-		},
-		IPv6Allocator: ipallocator.NewCIDRRange(nodeAddressing.IPv6().AllocationCIDR().IPNet),
+		IPv6Allocator:  ipallocator.NewCIDRRange(nodeAddressing.IPv6().AllocationCIDR().IPNet),
 	}
 
 	// Since docker doesn't support IPv6 only and there's always an IPv4
 	// address we can set up ipam for IPv4. More info:
 	// https://github.com/docker/libnetwork/pull/826
 	ipam.IPv4Allocator = ipallocator.NewCIDRRange(nodeAddressing.IPv4().AllocationCIDR().IPNet)
-	ipam.IPAMConfig.Routes = append(ipam.IPAMConfig.Routes,
-		// IPv4
-		&cniTypes.Route{
-			Dst: net.IPNet{
-				IP:   nodeAddressing.IPv4().Router(),
-				Mask: net.CIDRMask(32, 32),
-			},
-		},
-		&cniTypes.Route{
-			Dst: defaults.IPv4DefaultRoute,
```

Comment:
Where is the gateway defined now? Does this means that `github.com/containernetworking/plugins/plugins/ipam/host-local/backend/allocator` is not used anywhere?

Reply:
I didn't find any code using this. I will double check again.

Reply:
Removed the vendor code